### PR TITLE
Issue 2081 - 'hzn exchange org delete' agbot cleaning

### PR DESCRIPTION
**Pull request by sirotsinskuy**
Signed-off-by: Antonii Sirotsinskyi <asiro@softserveinc.com>

### CODE REVIEW

File: cli/exchange/org.go [hunk 1]
```diff
@@ -140,7 +140,7 @@ func OrgUpdate(org, userPwCreds, theOrg string, label string, desc string, tags
 
 	// check existance and also get current attribute values for comparizon later
 	var orgs exchange.GetOrganizationResponse
-	httpCode := cliutils.ExchangeGet("Exchange", cliutils.GetExchangeUrl(), "orgs/"+theOrg, cliutils.OrgAndCreds(org, userPwCreds), []int{200}, &orgs)
+	httpCode := cliutils.ExchangeGet("Exchange", cliutils.GetExchangeUrl(), "orgs/"+theOrg, cliutils.OrgAndCreds(org, userPwCreds), []int{200, 404}, &orgs)
```
Comment (linggao) [inline cli/exchange/org.go:143]: good catch. thanks.

File: cli/exchange/org.go [hunk 2]
```diff
@@ -195,15 +195,64 @@ func OrgDel(org, userPwCreds, theOrg string, force bool) {
...
-	httpCode := cliutils.ExchangeDelete("Exchange", cliutils.GetExchangeUrl(), "orgs/"+theOrg, cliutils.OrgAndCreds(org, userPwCreds), []int{204, 404})
+	// Search exchange for org, throw error if not found.
+	httpCode := cliutils.ExchangeGet("Exchange", cliutils.GetExchangeUrl(), "orgs/"+theOrg, cliutils.OrgAndCreds(org, userPwCreds), []int{200, 404}, nil)
 	if httpCode == 404 {
-		cliutils.Fatal(cliutils.NOT_FOUND, msgPrinter.Sprintf("org %s not found.", theOrg))
-	} else {
-		msgPrinter.Printf("Organization %v is successfully removed.", theOrg)
-		msgPrinter.Println()
+		cliutils.Fatal(cliutils.NOT_FOUND, msgPrinter.Sprintf("org '%s' not found.", theOrg))
+	}
```
Comment (linggao) [inline cli/exchange/org.go:202]: Please swap the org checking and "are you sure" confirmation part. If org is not found, no need to display confirmation.

File: cli/exchange/org.go [hunk 3]
```diff
@@ -195,15 +195,64 @@ func OrgDel(org, userPwCreds, theOrg string, force bool) {
...
+	// check if the agbot specified by -a exist or not
+	CheckAgbot(org, userPwCreds, agbot)
+
+	if agbot == "" {
+		// if -a is not specified, it go get the first agbot it can find
+		ag := GetDefaultAgbot(org, userPwCreds)
+		if ag == "" {
+			msgPrinter.Printf("No agbot found in the Exchange.")
+			msgPrinter.Println()
+			return
```
Comment (linggao) [inline cli/exchange/org.go:213]: It's okay there is no agbot found. Should not exit out. The code should skip the pattern and policy deletion part for agbot and proceed to delete the org.

File: cli/exchange/org.go [hunk 4]
```diff
@@ -195,15 +195,64 @@ func OrgDel(org, userPwCreds, theOrg string, force bool) {
...
+
+		}
+		agbot = ag
+		cliutils.Verbose(msgPrinter.Sprintf("Using agbot %v", agbot))
+	}
+
+	var agbotOrg string
+	agbotOrg, agbot = cliutils.TrimOrg(org, agbot)
+
+	// Load and remove all patterns
```
Comment (linggao) [inline cli/exchange/org.go:223]: Should be "Load and remove all agbot served patterns associated with this org"

File: cli/exchange/org.go [hunk 5]
```diff
@@ -195,15 +195,64 @@ func OrgDel(org, userPwCreds, theOrg string, force bool) {
...
+		patternJson := cliutils.MarshalIndent(p, "Cannot convert pattern to JSON")
+		cliutils.Unmarshal([]byte(patternJson), &servedPattern, msgPrinter.Sprintf("Cannot unmarshal served pattern"))
+
+		if servedPattern.PatternOrg == theOrg || servedPattern.NodeOrg == theOrg {
+			cliutils.Verbose(msgPrinter.Sprintf("Removing pattern %s from agbot %s", patternId, agbot))
+			cliutils.ExchangeDelete("Exchange", cliutils.GetExchangeUrl(), "orgs/"+agbotOrg+"/agbots/"+agbot+"/patterns/"+patternId, cliutils.OrgAndCreds(org, userPwCreds), []int{204})
+		}
+	}
+
+	// Load and remove all business policies
```
Comment (linggao) [inline cli/exchange/org.go:239]: Same here.

**Review by linggao — CHANGES_REQUESTED** [review]
per comments

File: cli/exchange/org.go (hunk 2 above)
Comment (sirotsinskuy) [reply cli/exchange/org.go:202]: What about abgot check (if it's provided by -a)? Should it remain after "are you sure" confirmation?

Comment (linggao) [reply cli/exchange/org.go:202]: Yes, CheckAgbot can also be put before "are you sure".

**Review by linggao — APPROVED** [review]

**Pull request merged**